src/client/app/reducers/formReducer.js

```diff
@@ -0,0 +1,68 @@
+import {
+    FORM_SUCCESS,
+    SUBMIT_ERROR,
+    SET_FORM_FIELDS,
+    UPDATE_FORM_FIELD,
+    SET_USER_AUTOCOMPLETE
+} from '../constants';
+
+const initialstate = {
+    errors: []
+};
+
+export default function formReducer (state = initialstate, action) {
+    switch (action.type) {
+
+    case SET_FORM_FIELDS:
+        const { fields, modelName, formType } = action.data;
+        return {
+            ...state,
```

Comment:
There's also an Object spread (see Babel plugin `transform-object-rest-spread`) in the ES7 spec. The return value in this case is the same as calling
```
Object.assign({}, state, {
    fields,
    modelName,
    formType
});
```
so, takes the current (form) state and merges those objects into it